content/articles/snorkel-python-for-labeling-datasets-programmatically/index.md

```diff
@@ -0,0 +1,395 @@
+
+The test set is used to evaluate the general performance of the model.
+It also checks if the model can make accurate predictions.
+
+Let's import the required method to split our dataset.
+
+```python
+from sklearn.model_selection import train_test_split
+```
+
+`train_test_split` will be used to split our dataset.
+
+```python
+df_train,df_test = train_test_split(df,train_size=0.5)
+```
+
+In the code above, we have specified the `train_size=0.5`. This implies that `50%` of the dataset will be used for training, and the remaining `50%` will be used for testing.
+
+Let's check the total number of sentences in our test set.
+
+```python
+print(df_train.shape)
+```
+
+The output is shown below.
+
+```bash
+(44, 1)
+```
+
+This shows we have a total of `44` sentences in our test set and `1` column.
+
+### Define our labeling functions
+
+Labelling functions define the rules that the labelling model uses. These rules are used to predict the label of unlabeled data.
+
+Let's import the method that will allow us to come up with labelling functions.
+
+```python
+from snorkel.labeling import labeling_function
+```
+
+So that we can come up with an accurate labelling model, we need to come up with at least three labelling functions.
+
+The imported `labeling_function` method allows us to create three labeling functions.
+
+To come up with the best labeling functions, we need to know how to label the dataset.
+
+In this case, we want to label our dataset with two labels as follows. A sentence can be labelled as either a question or a general statement. If the sentence is a question, it is labelled `1`, and a general statement is labelled `-1`.
+
+The following are rules for a statement to qualify to be a question.
+
+### Rules
+
+- A sentence should start with the following phrases: `why`, `what`, `when`, `who`, `where`, and `how`.
+- A sentence should end with a question mark, `?`.
+
+We then need to assign constants for our labels.
+
+#### Constants for our labels
+
+We will use this to label the sentences. For example, `QUESTION` is used to label sentences that qualify as questions.
+`ABSTAIN` will be used to label the other sentences.
+
+```python
+QUESTION = 1
+ABSTAIN = -1
+```
+
+Using the rules above, we can now come up with the labelling functions.
+
+### Keyword lookup function
+
+This function is used to check for phrases at the beginning of sentences.
+These phrases are as follows: `why`, `what`, `when`, `who`, `where`, and `how`.
+
+If this rule is met, the sentence is labelled as `QUESTION = 1`. If it's not met, the sentence is labelled `ABSTAIN = -1`.
+
+```python
+@labeling_function()
+def lf_keyword_lookup(x):
+  keywords = ["why","what","when","who","where","how"]
+ return QUESTION if any(word in x.sentences.lower() for word in keywords) else ABSTAIN
+```
+
+We have defined our function as `def lf_keyword_lookup(x)` with the help of the `@labeling_function()` method.
```

Comment:
The function name does not include def. Remove def when referring to `lf_keyword_lookup`. 

Secondly, @labeling_function is not a method. It is a decorator.